integration/src/test/scala/com/snowplowanalytics/snowplow/collectors/scalastream/intergation/KinesisSpec.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2013-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0, and
+ * you may not use this file except in compliance with the Apache License
+ * Version 2.0.  You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the Apache License Version 2.0 is distributed on an "AS
+ * IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the Apache License Version 2.0 for the specific language
+ * governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.collectors.scalastream.intergation
+
+import cats.effect.{IO, Sync}
+import cats.effect.testing.specs2.CatsIO
+import com.snowplowanalytics.snowplow.collectors.scalastream.intergation.TestUtils.Http.Request.RequestType.{Bad, Good}
+import com.snowplowanalytics.snowplow.collectors.scalastream.intergation.TestUtils._
+import com.snowplowanalytics.snowplow.eventgen.tracker.HttpRequest.Method.Post
+import org.specs2.mutable.Specification
+
+class KinesisSpec extends Specification with CatsIO {
+  "The Kinesis collector should" >> {
+    "ensure all events are written to the sink" in {
+      val localstack = Containers.localstack
+      val collector  = Containers.collector("kinesis", "config", Some(localstack))
+
+      lazy val localstackPort = Containers.getExposedPort(localstack, 4566)
+      lazy val collectorPort  = Containers.getExposedPort(collector, 12345)
+
+      val resources = for {
+        localstack <- Containers.mkContainer[IO](localstack)
+        collector  <- Containers.mkContainer[IO](collector)
+        kinesis    <- Kinesis.mkKinesisClient[IO](localstackPort)
+        httpClient <- Http.mkHttpClient[IO]
+      } yield (localstack, collector, kinesis, httpClient)
+
+      resources.use {
+        case (_, _, kinesis, httpClient) =>
+          val requestStubs = EventGenerator.makeStubs(10, 50)
+          val good         = requestStubs.map(Http.Request.make(_, collectorPort, Good))
+          val bad =
+            requestStubs
+              .filter(req =>
+                req.method match {
+                  case Post(_) if req.body.isDefined => true
+                  case _                             => false
+                }
+              )
+              .map(Http.Request.make(_, collectorPort, Bad))
+
+          val requests = good ++ bad
+
+          for {
+            _       <- Sync[IO].delay(println(s"Sending ${good.size} good and ${bad.size} bad events."))
+            _       <- Http.sendAll[IO](requests, httpClient)
+            _       <- Sync[IO].delay(Thread.sleep(10000)) // allow time for all records to be written before trying to read them
```

Comment:
```suggestion
            _       <- IO.sleep(10.seconds)
```

Reply:
Why is this better?

I mean, `Sync[IO].delay` is the same as `IO.apply`, so it's not a big deal. But it also means the `<-` needs to be changed to `=` just for this line, which irks me. Do you have a more important reason that just being concise?

Reply:
The reason is that `IO.sleep()` is exactly what you need and the most concise way. It's still to be used with `<-`.